CMake/resolve_dependency_modules/boost.cmake

```diff
@@ -24,3 +24,6 @@ set(Boost_NO_SYSTEM_PATHS ON)
 # We have to keep the FindBoost.cmake in an subfolder to prevent it from
 # overriding the system provided one when Boost_SOURCE=SYSTEM
 list(PREPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/boost)
+
+set(Boost_LIBRARIES headers;boost;atomic;context;date_time;filesystem;math;program_options;regex;system;thread)
+file(GLOB Boost_INCLUDE_DIRS "${CMAKE_CURRENT_BINARY_DIR}/_deps/boost-src/libs/*/include/")
```

Comment:
these are big hacks. I could find a good way to make fbthrift aware of boost libraries. It's still failing with:

```
: && /usr/lib64/ccache/c++ -fPIC -mavx2 -mfma -mavx -mf16c -mlzcnt -std=c++17 -mbmi2 -D USE_VELOX_COMMON_BASE -D HAS_UNCAUGHT_EXCEPTIONS -Wall -Wextra -Wno-unused        -Wno-unused-parameter        -Wno-sign-compare        -Wno-ignored-qualifiers        -Wno-implicit-fallthrough          -Wno-empty-body          -Wno-class-memaccess          -Wno-comment          -Wno-int-in-bool-context          -Wno-redundant-move          -Wno-array-bounds          -Wno-maybe-uninitialized          -Wno-unused-result          -Wno-format-overflow          -Wno-strict-aliasing          -Wno-type-limits          -Wno-stringop-overflow          -Wno-stringop-overread          -Wno-deprecated-declarations          -Wno-return-type -g  -Wl,-export-dynamic -shared -Wl,-soname,libcompiler_base.so.1.0.0 -o lib/libcompiler_base.so.1.0.0 _deps/fbthrift-build/thrift/compiler/CMakeFiles/compiler_base.dir/detail/system.cc.o _deps/fbthrift-build/thrift/compiler/CMakeFiles/compiler_base.dir/diagnostic.cc.o _deps/fbthrift-build/thrift/compiler/CMakeFiles/compiler_base.dir/source_location.cc.o _deps/fbthrift-build/thrift/compiler/CMakeFiles/compiler_base.dir/lib/schematizer.cc.o _deps/fbthrift-build/thrift/compiler/CMakeFiles/compiler_base.dir/parse/lexer.cc.o _deps/fbthrift-build/thrift/compiler/CMakeFiles/compiler_base.dir/parse/parser.cc.o _deps/fbthrift-build/thrift/compiler/CMakeFiles/compiler_base.dir/parse/parsing_driver.cc.o _deps/fbthrift-build/thrift/compiler/CMakeFiles/compiler_base.dir/parse/token.cc.o _deps/fbthrift-build/thrift/compiler/CMakeFiles/compiler_base.dir/sema/const_checker.cc.o _deps/fbthrift-build/thrift/compiler/CMakeFiles/compiler_base.dir/sema/patch_mutator.cc.o _deps/fbthrift-build/thrift/compiler/CMakeFiles/compiler_base.dir/sema/scope_validator.cc.o _deps/fbthrift-build/thrift/compiler/CMakeFiles/compiler_base.dir/sema/standard_mutator.cc.o _deps/fbthrift-build/thrift/compiler/CMakeFiles/compiler_base.dir/sema/standard_validator.cc.o  -Wl,-rpath,/home/pedroerp/Github/velox-fbthrift/_build/debug/lib  lib/libcompiler_ast.so.1.0.0  -lheaders  -lboost  -latomic  -lcontext  -ldate_time  -lfilesystem  -lmath  -lprogram_options  -lregex  -lsystem  -lthread  _deps/fmt-build/libfmtd.a  -lssl  -lcrypto && :
/usr/bin/ld: cannot find -lheaders
/usr/bin/ld: cannot find -lboost
/usr/bin/ld: cannot find -lcontext
/usr/bin/ld: cannot find -ldate_time
/usr/bin/ld: cannot find -lfilesystem
/usr/bin/ld: cannot find -lmath
/usr/bin/ld: cannot find -lprogram_options
/usr/bin/ld: cannot find -lregex
/usr/bin/ld: cannot find -lsystem
/usr/bin/ld: cannot find -lthread

```